pyuvdata/tests/test_uvdata.py

```diff
@@ -1548,3 +1548,366 @@ def test_get_feedpols():
+    ant_str = '(1,3)_11'
+    ant_pairs = [(1, 11), (3, 11)]
+    uv2 = uv.select(ant_str=ant_str, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), ant_pairs)
+    nt.assert_items_equal(uv2.get_pols(), uv.get_pols())
+
+    # Specific baselines with parenthesis
+    ant_str = '1_(3,11)'
+    ant_pairs = [(1, 3), (1, 11)]
+    uv2 = uv.select(ant_str=ant_str, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), ant_pairs)
+    nt.assert_items_equal(uv2.get_pols(), uv.get_pols())
+
+    # Antenna numbers with polarizations
+    ant_str = '(1l,2r)_(3l,6r)'
+    ant_pairs = [(1, 3), (1, 6), (2, 3), (2, 6)]
+    pols = ['RR', 'LL', 'RL', 'LR']
+    uv2 = uv.select(ant_str=ant_str, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), ant_pairs)
+    nt.assert_items_equal(uv2.get_pols(), pols)
+
+    # Antenna numbers with - for avoidance
+    ant_str = '1_(-3,11)'
+    ant_pairs = [(1, 11)]
+    uv2 = uv.select(ant_str=ant_str, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), ant_pairs)
+    nt.assert_items_equal(uv2.get_pols(), uv.get_pols())
+
+    ant_str = '(-1,3)_11'
+    ant_pairs = [(3, 11)]
+    uv2 = uv.select(ant_str=ant_str, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), ant_pairs)
+    nt.assert_items_equal(uv2.get_pols(), uv.get_pols())
+
+    # Remove specific antenna number
+    ant_str = '1,-3'
+    ant_pairs = [(0, 1), (1, 2), (1, 6), (1, 7), (1, 8), (1, 11),
+                       (1, 14), (1, 18), (1, 19), (1, 20), (1, 21),
+                       (1, 22), (1, 23), (1, 24), (1, 26), (1, 27)]
+    uv2 = uv.select(ant_str=ant_str, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), ant_pairs)
+    nt.assert_items_equal(uv2.get_pols(), uv.get_pols())
+
+    # Remove specific baseline
+    ant_str = '1,-1_3'
+    ant_pairs = [(0, 1), (1, 2), (1, 6), (1, 7), (1, 8), (1, 11),
+                       (1, 14), (1, 18), (1, 19), (1, 20), (1, 21),
+                       (1, 22), (1, 23), (1, 24), (1, 26), (1, 27)]
+    uv2 = uv.select(ant_str=ant_str, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), ant_pairs)
+    nt.assert_items_equal(uv2.get_pols(), uv.get_pols())
+
+    # Antenna numbers with polarizations and - for avoidance
+    ant_str = '1l_(-3r,11l)'
+    ant_pairs = [(1, 11)]
+    pols = ['LL']
+    uv2 = uv.select(ant_str=ant_str, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), ant_pairs)
+    nt.assert_items_equal(uv2.get_pols(), pols)
+
+    # Pass ant_str with ant_pairs_nums and polarizations to select
+    ant_str = '1l_3r'
+    ant_pairs_nums = [(1, 11)]
+    ant_pairs = [(1, 3), (1, 11)]
+    polarizations = [-1] # RR
+    pols = ['RR', 'LR']
+    uv2 = uv.select(ant_str=ant_str, ant_pairs_nums=ant_pairs_nums,
+                            polarizations=polarizations, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), ant_pairs)
+    nt.assert_items_equal(uv2.get_pols(), pols)
```

Comment:
This last test is great! You should probably also have test for the case where ant_str is set along with antenna_numbers or antenna_names.